### ISSUE TITLE
Finding selectors for Jira?

### ISSUE THREAD

**Issue by mbourgon** [zephyraft/mermaid-previewer#83]
Howdy. This works great in Confluence but I can't figure out what selector I need to use for Jira. I'm not a web person, but inspecting a page where I added mermaid, I found div.code.panel, div.codecontent.panelcontent, div.user-content-block, and div.pre.code-java (I tried adding it as a default code block, which I guess is java). How would I try those in the selector? I tried "pre > code-java" and "div.user-content-block > p", but neither worked. 

Here's what that part looks like in the inspector...
<div class="user-content-block">
<div class="code panel" style="border-width: 1px;">
<div class="codeContent panelContent">
<pre class="code-java"> == $0
"```mermaid flowchart LR; A["
(and then the mermaid lines)
(and then these lines exiting out of that code block, so we're in the main user-content-block)
</div>
</div>
<p>== $0
"```mermaid"
(and then another block of mermaid, this time not in a code block)

**Comment by zephyraft** [comment]
jira and confluence code blocks cannot select the mermaid language, so can only use None or PlainText instead.

<img width="817" alt="image" src="https://github.com/zephyraft/mermaid-previewer/assets/24712991/aab066f1-faa2-4fe6-a3c9-92e63e9464c1">


Perhaps you can try adding the selector **code.language-text** to the **atlassian.net** domain.
<img width="1119" alt="image" src="https://github.com/zephyraft/mermaid-previewer/assets/24712991/b86379ba-e2a9-4c80-9575-c00588b56d58">

Then refresh the page, you can see that it has taken effect.
<img width="807" alt="image" src="https://github.com/zephyraft/mermaid-previewer/assets/24712991/a2531961-4b2b-46fe-9e0a-0bfdd33c3bb5">

**Comment by zephyraft** [comment]
Using the above configuration, it works fine in Confluence, but there seems to be a styling issue in Jira's boards.

<img width="483" alt="image" src="https://github.com/zephyraft/mermaid-previewer/assets/24712991/71eca026-02f6-49da-8844-399b361b3b0e">

<img width="486" alt="image" src="https://github.com/zephyraft/mermaid-previewer/assets/24712991/bba497e9-cbfe-4b92-86e2-c78c65c86bdd">

**Comment by mbourgon** [comment]
Thanks for looking.  Oddly, for Confluence what I figured out what that using markdown code blocks works, using "pre > code". (That's pretty easy to set up: "{", then "markd", then [enter]).  What you provided doesn't work on our Confluence (self-hosted?); I tried that, tried td.code, tried div.codeContent. So there are options for confluence, but yeah, Jira is a PITA. I appreciate you looking!

**Comment by zephyraft** [comment]
> Using the above configuration, it works fine in Confluence, but there seems to be a styling issue in Jira's boards.使用上述配置，它在 Confluence 中工作正常，但 Jira 的面板中似乎存在样式问题。
> 
> <img alt="image" width="483" src="https://user-images.githubusercontent.com/24712991/273423192-71eca026-02f6-49da-8844-399b361b3b0e.png"> <img alt="image" width="486" src="https://user-images.githubusercontent.com/24712991/273423197-bba497e9-cbfe-4b92-86e2-c78c65c86bdd.png">

The newly released version, v1.4.4, added an experimental feature called Sandbox mode, which can solve the styling issues of the open-source version of Jira. (You can enable it through the settings page.)

<img width="502" alt="image" src="https://github.com/zephyraft/mermaid-previewer/assets/24712991/8aca4e22-3b36-487f-b0b0-b688743ab99a">


For a self-hosted Jira, the required selector configuration may vary depending on the version. You can use Chrome Developer Tools to locate the specific DOM element and find the corresponding selector.

<img width="971" alt="image" src="https://github.com/zephyraft/mermaid-previewer/assets/24712991/df2190b8-d1f0-4227-b34d-9b675a7eead6">

**Comment by mbourgon** [comment]
GOT IT!

So when I create a code block in our Jira, the selector says pre.code-java.  I created a selector "pre.code-java" for our Jira site, and it's more finicky than I expected (the block can ONLY contain "graph LR" and "A --> B", with that precise spacing), but it works.  

And I just found a weird difference in how I'm doing it. Originally because I misread things (how the three tick marks work). 
On Confluence, I do this inside a markdown block, and it works:
[markdown block begins]
```mermaid 
graph LR
A-->B
```
[markdown block ends]
(and if I do it without the three tick marks and mermaid, it doesn't resolve)

but in Jira, I have to do JUST:
```
graph LR
A --> B

```


Jira doesn't work: 
```mermaid
graph LR
A-->B
```

Jira doesn't work:
mermaid
graph LR
A --> B

Jira works:
graph LR
A --> B
